rust/noosphere-gateway/src/worker/name_system.rs

```diff
@@ -376,3 +437,60 @@ impl<H> OnDemandNameResolver<H> {
         Ok(rx.await?)
     }
 }
+
+async fn set_counterpart_record<C, K, S>(context: C, record: &Jwt) -> Result<()>
+where
+    C: HasMutableSphereContext<K, S>,
+    K: KeyMaterial + Clone + 'static,
+    S: Storage + 'static,
+{
+    debug!("Setting counterpart record...");
+    let counterpart_identity = {
+        let sphere_context = context.sphere_context().await?;
+        let db = sphere_context.db();
+        db.require_key::<_, Did>(COUNTERPART).await?
+    };
+    let counterpart_link_record_key = format!("link_record/{counterpart_identity}");
+    let mut cursor = SphereCursor::latest(context.clone());
+    let (_, bytes) = block_serialize::<DagCborCodec, _>(record)?;
+
+    cursor
+        .write(
+            &counterpart_link_record_key,
+            &ContentType::Cbor.to_string(),
+            Cursor::new(bytes),
```

Comment:
TBH if you are just storing the JWT, the CBOR encoding is kind of overkill. Might as well just store UTF-8 string bytes :shrug: 

Encoding as CBOR is nice for structured data. But, `Jwt` is just a wrapper around a `String`. So...

```rs
cursor.write(
    slug,
    content_type,
    record.as_bytes(),
    None,
).await?;
```

...should suffice. Then, when reading:

```rs
let mut buffer = String::new();
let mut file = context.read(...).await?.unwrap_or_else(...)?;
file.context.read_to_string(&mut buffer).await?;
let jwt = Jwt(buffer);
```

Reply:
If we implemented `DerefMut` for our string wrappers we could even pass a mutable `Jwt` instead of a naked string when reading.... :thinking: